docs/rules.md

```diff
@@ -0,0 +1,11 @@
+# Rules in test runners
+
+## Tests must be run in order.
```

Comment:
This is sensible from the point of view of encouraging TDD, but we should remain aware that the total test execution time (and therefore the total bill for running the test suites) will be greater when running tests sequentially than when running them in parallel.

Reply:
For some test libraries this might be impossible.

As far as I remember, there is no "run tests in order" for ruby. And for elixir its only available indirectly, as a randomizer seed of `0` will not shuffle the tests, but this is an implementation detail and even though widely known, not guaranteed to hold for ever.

Reply:
>As far as I remember, there is no "run tests in order" for ruby. 

For minitest they're run in order by default. For other Ruby libraries this might be different.

>  And for elixir its only available indirectly, as a randomizer seed of 0 will not shuffle the tests, but this is an implementation detail and even though widely known, not guaranteed to hold for ever.

For these cases, I guess we'll need to do something like emit the line-number of any tests that fail and then take the lowest one, or something?

Reply:
Line numbers seem like a good thing to include anyway I feel.

Reply:
Ah. Pretty sure that's changed since I first used minitest! :) Thanks for finding the docs, @kotp

Reply:
It changed in 2013, from what I can tell superficially.

Reply:
I'm moving this out of this PR and into https://github.com/exercism/automated-tests/pull/19